learningequality/kolibri
# [GSoC] Use KTable in Facility -> Classes

## Summary
This PR introduces `KTable` component to Kolibri as a part of GSoC Project : Accessible Sortable Table. 
The component is still under review in [KDS](https://github.com/learningequality/kolibri-design-system/pull/727).


## Reviewer guidance

- Navigate to [root]/en/facility/#/classes
- Sign in if prompted
- Click on the `New Class` button
- Add a random class. Once added, the information is displayed using KTable on the classes page.
- Add a few more random classes using `New Class` button to check functionalities of the table component.

## QA Testing Guidance
- Table navigable using both arrow keys and tab key.
- When navigating any cell using arrow key or tab key, the corresponding column header and entire row containing the cell are highlighted.
- Elements within cells that are focusable, receive proper focus using tab key and can be executed using 'Enter' key
- The headers can be sorted by pressing 'Enter' key on them
- The focus ring is visible clearly around the focused elements.
- When viewport is shrinked for smaller devices, navigation using arrow keys/tab key should ensure that each element is clearly visible when navigating through them and does not get hidden by sticky headers.


## Testing checklist



## PR process


## Reviewer checklist

- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md
